### DISCUSSION TITLE
Sharing actor references across RPC boundary

### DISCUSSION THREAD

**Discussion by SuddenlyHazel** [slawlor/ractor#178] [Q&A]
Hi all,

I'm trying to create a distributed actor directory and I'm hitting some issues with `BytesConvertable` and `ActorRef`.
It's pretty clear now I can't send `ActorRef` or `ActorCell` over the wire. Does anyone have any suggestions for how I can work around this?

Thanks!

My message is as follows if it helps

```rust
#[derive(RactorClusterMessage)]
pub enum DirectoryMessage {
    Put(
        Domain,
        ActorRef<http_function_actor::HttpFunctionActorMessage>,
    ),
    Get(
        Domain,
        RpcReplyPort<Option<ActorRef<http_function_actor::HttpFunctionActorMessage>>>,
    )
}
```

**Comment by slawlor** [comment]
Unfortunately the best I can offer at this point is to send the local ActorId as the payload, which is serializable, and then you would use the remote server's pid registry to lookup the actor to get an ActorRef to reply to. This is more akin to what Erlang does, as they just send actor "addresses" around. However this means you'll need to deal with that syntax somewhat manually :(

If you have any ideas or suggestions, I'd be happy to hear them. I can pretty much assure however that an ActorRef/Cell won't be every serializable given they hold an in-memory channel representing the actor's communication mailbox. This isn't something that can be split over a network bridge.

**Comment by slawlor** [comment]
Also fully open to PRs with enhanced functionality if you so choose!
